app/Database/Migrations/2024-09-27-171916_AddSocialMediaLink.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace App\Database\Migrations;
+
+use CodeIgniter\Database\Migration;
+
+class AddSocialMediaLink extends Migration
+{
+    public function up()
+    {
+        $this->forge->addField([
+            'id' => [
+                'type' => 'INT',
+                'unsigned' => true,
+                'auto_increment' => true,
+            ],
+            'social_media_type_id' => [
+                'type' => 'INT',
+                'unsigned' => true,
+                'null' => false,
+            ],
+            'speaker_id' => [
```

Comment:
The speaker ID stays the same when updating the speaker right?
Do we generate a speaker per year so that the links stay in a event or do the old events update too when a speaker updates their links.

We could add a event_year here to make clear to with events this links belong.

im not quite sure what we want here.

Reply:
No, the speaker ID doesn't stay the same. If a speaker makes changes to their profile (e.g. the description), a new speaker entry will be created. This also means that all social media links have to be copied. The copying and the amount of resulting database entries is not an issue here. But what *could* be an issue: If a speaker changes a social media link for, say, year 2025, the corresponding link will *not* be updated for 2024.
We have to decide whether this is intended or not.

An alternative approach would be to "connect" the social media links to the user ID instead of the speaker ID. That would mean that the social media links for a person are identical for all years and changes to the links will be reflected to the old years as well.

Which one do you prefer?

Reply:
I think i prefer to link the links to the user id.
There is no point to have an old link somewhere.
Its just not working, right?
So it would be better to have only one set of links.
When I kick on an older Link it should still work right?

Reply:
I'm with @PurpurTentakel97 on this one. 😄